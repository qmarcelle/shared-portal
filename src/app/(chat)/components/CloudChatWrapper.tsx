'use client';

// CloudChatWrapper injects the Genesys Cloud Messenger SDK and manages Messenger-specific events.
// All script loading, Messenger events, and errors are logged for traceability and debugging.

import { useEffect } from 'react';
import { useChatStore } from '../stores/chatStore';

export default function CloudChatWrapper() {
  const { userData } = useChatStore();

  useEffect(() => {
    // Build chatSettings from env and userData
    window.chatSettings = {
      bootstrapUrl: process.env.NEXT_PUBLIC_GENESYS_BOOTSTRAP_URL || '',
      widgetUrl: process.env.NEXT_PUBLIC_GENESYS_WIDGET_URL || '',
      clickToChatJs: process.env.NEXT_PUBLIC_GENESYS_CLICK_TO_CHAT_JS || '',
      clickToChatEndpoint: process.env.NEXT_PUBLIC_CLICK_TO_CHAT_ENDPOINT || '',
      chatTokenEndpoint: process.env.NEXT_PUBLIC_CHAT_TOKEN_ENDPOINT || '',
      coBrowseEndpoint: process.env.NEXT_PUBLIC_COBROWSE_LICENSE_ENDPOINT || '',
      opsPhone: process.env.NEXT_PUBLIC_OPS_PHONE || '',
      opsPhoneHours: process.env.NEXT_PUBLIC_OPS_HOURS || '',
      ...userData,
    };
    console.debug('[CloudChatWrapper] chatSettings:', window.chatSettings);
  }, [userData]);

  useEffect(() => {
    // Load Genesys Cloud bootstrap and widget scripts
    const bootstrapUrl = window.chatSettings?.bootstrapUrl || '';
    const widgetUrl = window.chatSettings?.widgetUrl || '';
    if (!bootstrapUrl) return;
    const bootstrapScript = document.createElement('script');
    bootstrapScript.src = bootstrapUrl;
    bootstrapScript.async = true;
    bootstrapScript.onload = () => {
      console.debug('[CloudChatWrapper] Genesys bootstrap loaded');
      if (!widgetUrl) return;
      const widgetScript = document.createElement('script');
      widgetScript.src = widgetUrl;
      widgetScript.async = true;
      widgetScript.onload = () => {
        console.debug('[CloudChatWrapper] Genesys widgets loaded');
      };
      widgetScript.onerror = (e) => {
        console.error('[CloudChatWrapper] Failed to load widgets script', e);
      };
      document.body.appendChild(widgetScript);
    };
    bootstrapScript.onerror = (e) => {
      console.error('[CloudChatWrapper] Failed to load bootstrap script', e);
    };
    document.body.appendChild(bootstrapScript);

    // Provide openGenesysChat logic
    window.openGenesysChat = () => {
      if (window.Genesys) {
        try {
          window.Genesys('command', 'Messenger.open');
        } catch (e) {
          console.error(
            '[CloudChatWrapper] Error opening Genesys Messenger',
            e,
          );
        }
      }
    };

    return () => {
      // Cleanup scripts and chatSettings
      document.body.removeChild(bootstrapScript);
      if (widgetUrl) {
        const widgetScript = document.querySelector(
          'script[src="' + widgetUrl + '"]',
        );
        if (widgetScript) document.body.removeChild(widgetScript);
      }
      if (window.chatSettings) delete window.chatSettings;
      if (window.openGenesysChat) delete window.openGenesysChat;
      console.debug('[CloudChatWrapper] Cleanup complete');
    };
  }, [userData]);

  return null;
}
