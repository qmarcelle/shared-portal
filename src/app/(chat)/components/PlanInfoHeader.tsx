'use client';
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { Title } from '@/components/foundation/Title';

/**
 * Show current plan when multiple plans exist
 */
export default function PlanInfoHeader() {
  const { formInputs } = useChatStore();
  if (formInputs.length <= 1) return null;
  const plan = formInputs.find((f) => f.id === 'PLAN_ID');

  return (
    <Title
      className="plan-info-header"
      text={`You are chatting about plan ${plan?.value || ''}`}
    />
  );
}
