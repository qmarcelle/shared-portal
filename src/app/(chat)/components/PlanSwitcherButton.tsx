'use client';
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { Button } from '@/components/foundation/Button';
import { usePlanStore } from '../../../userManagement/stores/planStore';

/**
 * "Switch Plan" inside pre-chat window
 */
export default function PlanSwitcherButton() {
  const { endChat } = useChatStore();
  const { plans, selectedPlanId, openPlanSwitcher } = usePlanStore();

  if (plans.length <= 1) return null;

  const handleSwitch = () => {
    endChat();
    openPlanSwitcher();
  };

  return (
    <Button
      label="Switch Plan"
      callback={handleSwitch}
      className="plan-switch-button"
      type="secondary"
    />
  );
}
