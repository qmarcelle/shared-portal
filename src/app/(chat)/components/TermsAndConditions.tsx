'use client';
// import TCs from '@/app/@chat/config/terms'; // { default: '…', lobA: '…', … }
import { useChatStore } from '@/app/(chat)/stores/chatStore';
import { RichText } from '@/components/foundation/RichText';

/**
 * Render T&C based on LOB
 */
export default function TermsAndConditions() {
  const { userData } = useChatStore();
  // const text = TCs[userData.LOB] || TCs.default;
  return (
    <RichText
      spans={['Terms and Conditions']}
      className="terms-and-conditions"
      type="body-2"
    />
  );
}
