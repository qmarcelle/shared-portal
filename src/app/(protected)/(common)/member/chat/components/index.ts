/**
 * Component Exports
 * Centralizes all component exports following the new consolidated structure
 */

// Main Components
export { ChatTrigger } from './ChatTrigger';
export { ChatWidget } from './ChatWidget';
export { PlanInfoHeader } from './PlanInfoHeader';

// Error Handling
export { ChatErrorBoundary } from './shared/ChatErrorBoundary';

// Integration Components (to be consolidated)
export { GenesysScripts } from './GenesysScripts';
