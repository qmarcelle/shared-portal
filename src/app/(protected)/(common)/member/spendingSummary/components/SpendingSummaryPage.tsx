import { FilterDetails } from '@/models/filter_dropdown_details';
import SpendingSummary from '..';
import { getInitialClaimsFilter } from '../actions/getInitialFilter';
import { SpendingType } from '../models/SpendingType';

interface SpendingSummaryPageProps {
  spendingType: SpendingType;
}

export function SpendingSummaryPage({
  spendingType,
}: SpendingSummaryPageProps) {
  const filterItems = getInitialClaimsFilter();
  filterItems[0].value = [...(filterItems[0].value! as FilterDetails[])];

  const filterVal = (filterItems[2].value as FilterDetails[]).find(
    (item: FilterDetails): boolean =>
      item.value.toLowerCase() === spendingType.toLowerCase(),
  );

  if (filterVal) {
    filterItems[2].selectedValue = filterVal;
  }

  return <SpendingSummary filters={filterItems} />;
}
