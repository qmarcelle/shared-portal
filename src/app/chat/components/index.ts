// Core Components
export * from './core/ChatButton';
export * from './core/ChatErrorBoundary';
export * from './core/ChatHeader';
export * from './core/ChatWidget';

// Business Hours Components
export * from './business-hours/BusinessHoursNotification';

// Eligibility Components
export * from './eligibility/EligibilityCheck';

// Plan Switcher Components will be added here when implemented
// export * from './plan-switcher/PlanSwitcher';
