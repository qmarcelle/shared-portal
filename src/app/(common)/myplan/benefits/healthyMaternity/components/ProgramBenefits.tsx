/* eslint-disable @typescript-eslint/no-unused-vars */
/* eslint-disable react/jsx-key */
import { Column } from '@/components/foundation/Column';
import { Header } from '@/components/foundation/Header';
import { Row } from '@/components/foundation/Row';
import { Spacer } from '@/components/foundation/Spacer';
import { IComponent } from '../../../components/IComponent';
import { TextBox } from '../../../components/foundation/TextBox';
import { BenefitDetails } from '../models/benefit_details';

interface ProgramBenefitsProps extends IComponent {
  benefits: BenefitDetails[];
}

export const ProgramBenefits = ({ benefits }: ProgramBenefitsProps) => {
  return (
    <>
      <Spacer size={12} />
      <Header className="title-2 px-3" text="Program Benefits"></Header>
      <Spacer size={25} />
      <Row className="hidden sm:flex px-3">
        {benefits.slice(0, benefits.length).map((item, index) => (
          <Column className="md:pr-4 md:w-1/3">
            {item.benefitIcon}
            <Spacer size={5} />
            <TextBox text={item.benefitLabel} className="font-bold" />
            <Spacer size={5} />
            <TextBox text={item.benefitCopy} />
          </Column>
        ))}
      </Row>
      <Column className="block sm:hidden px-3">
        {benefits.slice(0, benefits.length).map((item, index) => (
          <Column className="pr-4 mb-4">
            {item.benefitIcon}
            <Spacer size={5} />
            <TextBox text={item.benefitLabel} className="font-bold" />
            <Spacer size={5} />
            <TextBox text={item.benefitCopy} />
          </Column>
        ))}
      </Column>
    </>
  );
};
