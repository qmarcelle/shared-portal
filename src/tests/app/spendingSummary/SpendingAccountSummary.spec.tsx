import SpendingSummaryPage from '@/app/spendingSummary/page';
import {
  dentalIcon,
  medicalIcon,
  pharmacyIcon,
  visionIcon,
} from '@/components/foundation/Icons';
import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import Image from 'next/image';
import { AnnualSpendingSummary } from '../../../app/dashboard/components/AnnualSpendingSummary';

const renderUI = () => {
  return render(
    <AnnualSpendingSummary
      className="large-section statementSummary"
      title="Statement Summary up to November 8, 2023"
      subTitle="View Medical, Pharmacy, Dental and Vision for All Members"
      amountPaid={1199.19}
      totalBilledAmount={9804.31}
      amountSaved={8605.12}
      amountSavedPercentage={89}
      color1={'#005EB9'}
      color2={'#5DC1FD'}
      service={[
        {
          serviceIcon: (
            <Image className="w-6" src={medicalIcon} alt="Medical Icon" />
          ),
          serviceLabel: 'Medical',
          serviceSubLabel: 'Your share',
          serviceSubLabelValue: 30.24,
          labelText1: 'Amount Billed',
          labelValue1: 145.0,
          labelText2: 'Plan Discount',
          labelValue2: 114.76,
          labelText3: 'Plan Paid',
          labelValue3: 0.0,
        },
        {
          serviceIcon: (
            <Image className="w-6" src={pharmacyIcon} alt="Pharmacy Icon" />
          ),
          serviceLabel: 'Pharmacy',
          serviceSubLabel: 'Your share',
          serviceSubLabelValue: 30.24,
          labelText1: 'Amount Billed',
          labelValue1: 145.0,
          labelText2: 'Plan Discount',
          labelValue2: 114.76,
          labelText3: 'Plan Paid',
          labelValue3: 0.0,
        },
        {
          serviceIcon: (
            <Image className="w-6" src={dentalIcon} alt="Dental Icon" />
          ),
          serviceLabel: 'Dental',
          serviceSubLabel: 'Your share',
          serviceSubLabelValue: 30.24,
          labelText1: 'Amount Billed',
          labelValue1: 145.0,
          labelText2: 'Plan Discount',
          labelValue2: 114.76,
          labelText3: 'Plan Paid',
          labelValue3: 0.0,
        },
        {
          serviceIcon: (
            <Image className="w-6" src={visionIcon} alt="Vision Icon" />
          ),
          serviceLabel: 'Vision',
          serviceSubLabel: 'Your share',
          serviceSubLabelValue: 30.24,
          labelText1: 'Amount Billed',
          labelValue1: 145.0,
          labelText2: 'Plan Discount',
          labelValue2: 114.76,
          labelText3: 'Plan Paid',
          labelValue3: 0.0,
        },
      ]}
    />,
  );
};

describe('AnnualSpendingSummary', () => {
  it('should render the UI correctly', async () => {
    const component = renderUI();
    expect(
      screen.getByText('Statement Summary up to November 8, 2023'),
    ).toBeVisible();
    expect(screen.getByText('You Paid')).toBeVisible();
    expect(screen.getByText('You Saved')).toBeVisible();
    expect(screen.getByText('Your plan paid')).toBeVisible;
    expect(screen.getByText('of your costs')).toBeVisible();
    expect(screen.getByText('Total Billed')).toBeVisible();

    expect(screen.getByText('Medical')).toBeInTheDocument();
    expect(screen.getByText('Pharmacy')).toBeInTheDocument();
    expect(screen.getByText('Dental')).toBeInTheDocument();
    expect(screen.getByText('Vision')).toBeInTheDocument();

    let serviceRenderedSectionContent = screen.queryByText('Amount Billed');
    expect(serviceRenderedSectionContent).not.toBeInTheDocument();

    fireEvent.click(screen.getByText('Medical')); //check that body text is visible after clicking
    serviceRenderedSectionContent = screen.getByText('Amount Billed');
    expect(serviceRenderedSectionContent).toBeInTheDocument();

    fireEvent.click(screen.getByText('Medical')); //check that it is not visible after clicking again
    serviceRenderedSectionContent = screen.queryByText('Amount Billed');
    expect(serviceRenderedSectionContent).not.toBeInTheDocument();

    expect(component.baseElement).toMatchSnapshot();
  });

  it('should render the UI correctly', async () => {
    const Result = await SpendingSummaryPage({
      searchParams: new Promise((resolve) => {
        resolve({ type: 'Pharmacy' });
      }),
    });
    const component = render(Result);
    screen.getByRole('heading', {
      name: 'Spending Summary',
    });
    screen.getByRole('button', {
      name: /Download a PDF Statement/i,
    });
    screen.getAllByText('Pharmacy');
    expect(component).toMatchSnapshot();
  });
});
