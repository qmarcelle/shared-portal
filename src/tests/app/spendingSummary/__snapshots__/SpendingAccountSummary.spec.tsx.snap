// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpendingAccountSummary should render the UI correctly 1`] = `
<body>
  <div>
    <div
      class="card-main large-section statementSummary"
    >
      <div>
        <h2
          class="title-2"
        >
          Statement Summary up to November 8, 2023
        </h2>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
        />
        <div
          class="flex flex-row"
        >
          <p
            class="body-1"
          >
            As of
          </p>
          <span
            style="display: block; width: 8px; min-width: 8px; height: 1px; min-height: 1px;"
          />
          <p>
            View Medical, Pharmacy, Dental and Vision for All Members
          </p>
        </div>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
        />
        <div
          class="flex flex-col chartSection"
        >
          <div
            class="text-center absolute doughnutText pl-3 w-36"
          >
            <div
              class="flex flex-col justify-between pl-3 chartText"
            >
              <p>
                Your plan paid
              </p>
              <p
                class="font-bold"
              >
                89
                %
              </p>
              <p>
                of your costs
              </p>
            </div>
            <span
              style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
            />
            <div
              class="chartTextDivider"
            >
              <div
                aria-label="divider"
                class=""
                style="width: 100%; height: 1px; background-color: rgb(204, 204, 204);"
              />
            </div>
            <span
              style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
            />
            <div
              class="flex flex-col justify-between pl-3 chartText"
            >
              <p>
                Total Billed
              </p>
              <p
                class="font-bold"
              >
                $9,804.31
              </p>
            </div>
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 8px; min-height: 8px;"
          />
          <canvas
            class="spendingSummaryChart"
            height="150"
            role="img"
            width="300"
          />
          <span
            style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
          />
        </div>
        <div
          class="sectionAlignment"
        >
          <div
            class="flex flex-row"
          >
            <div
              style="width: 20px; height: 20px; background-color: rgb(0, 94, 185); border-radius: 50px; margin-top: 0px; margin-right: 0px;"
            />
            <p
              class="body-1 ml-2"
            >
              You Paid
            </p>
            <p
              class="font-bold ml-auto"
            >
              $1,199.19
            </p>
          </div>
          <span
            style="display: block; width: 1px; min-width: 1px; height: 18px; min-height: 18px;"
          />
          <div
            class="flex flex-row"
          >
            <div
              style="width: 20px; height: 20px; background-color: rgb(93, 193, 253); border-radius: 50px; margin-top: 0px; margin-right: 0px;"
            />
            <p
              class="body-1 ml-2"
            >
              You Saved
            </p>
            <p
              class="font-bold ml-auto"
            >
              $8,605.12
            </p>
          </div>
        </div>
        <span
          style="display: block; width: 1px; min-width: 1px; height: 32px; min-height: 32px;"
        />
        <div
          class="flex flex-col clear-left"
        >
          <div
            class="card-main mb-3"
          >
            <div
              class="flex flex-col items-stretch"
            >
              <div
                class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4"
              >
                <div
                  class="flex flex-row m-1 accordion-head"
                >
                  <div
                    class="mr-1"
                  >
                    <img
                      alt="Medical Icon"
                      class="w-6"
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                  </div>
                  <p
                    class="mr-2 flex-grow"
                  >
                    Medical
                  </p>
                  <img
                    alt="Down Chevron"
                    class="pl-2 w-6"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                </div>
                <div
                  class="mr-2 flex-grow"
                >
                  <div
                    class="flex flex-row m-1 justify-between"
                  >
                    <p
                      class="body-1 font-bold"
                    >
                      Your share
                    </p>
                    <p
                      class="body-1 font-bold"
                    >
                      $30.24
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card-main mb-3"
          >
            <div
              class="flex flex-col items-stretch"
            >
              <div
                class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4"
              >
                <div
                  class="flex flex-row m-1 accordion-head"
                >
                  <div
                    class="mr-1"
                  >
                    <img
                      alt="Pharmacy Icon"
                      class="w-6"
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                  </div>
                  <p
                    class="mr-2 flex-grow"
                  >
                    Pharmacy
                  </p>
                  <img
                    alt="Down Chevron"
                    class="pl-2 w-6"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                </div>
                <div
                  class="mr-2 flex-grow"
                >
                  <div
                    class="flex flex-row m-1 justify-between"
                  >
                    <p
                      class="body-1 font-bold"
                    >
                      Your share
                    </p>
                    <p
                      class="body-1 font-bold"
                    >
                      $30.24
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card-main mb-3"
          >
            <div
              class="flex flex-col items-stretch"
            >
              <div
                class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4"
              >
                <div
                  class="flex flex-row m-1 accordion-head"
                >
                  <div
                    class="mr-1"
                  >
                    <img
                      alt="Dental Icon"
                      class="w-6"
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                  </div>
                  <p
                    class="mr-2 flex-grow"
                  >
                    Dental
                  </p>
                  <img
                    alt="Down Chevron"
                    class="pl-2 w-6"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                </div>
                <div
                  class="mr-2 flex-grow"
                >
                  <div
                    class="flex flex-row m-1 justify-between"
                  >
                    <p
                      class="body-1 font-bold"
                    >
                      Your share
                    </p>
                    <p
                      class="body-1 font-bold"
                    >
                      $30.24
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card-main mb-3"
          >
            <div
              class="flex flex-col items-stretch"
            >
              <div
                class="flex flex-col card-main-elevated p-1 cursor-pointer px-2 py-4"
              >
                <div
                  class="flex flex-row m-1 accordion-head"
                >
                  <div
                    class="mr-1"
                  >
                    <img
                      alt="Vision Icon"
                      class="w-6"
                      data-nimg="1"
                      decoding="async"
                      height="40"
                      loading="lazy"
                      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                      style="color: transparent;"
                      width="40"
                    />
                  </div>
                  <p
                    class="mr-2 flex-grow"
                  >
                    Vision
                  </p>
                  <img
                    alt="Down Chevron"
                    class="pl-2 w-6"
                    data-nimg="1"
                    decoding="async"
                    height="40"
                    loading="lazy"
                    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
                    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
                    style="color: transparent;"
                    width="40"
                  />
                </div>
                <div
                  class="mr-2 flex-grow"
                >
                  <div
                    class="flex flex-row m-1 justify-between"
                  >
                    <p
                      class="body-1 font-bold"
                    >
                      Your share
                    </p>
                    <p
                      class="body-1 font-bold"
                    >
                      $30.24
                    </p>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
